tests/jerry/regression-test-issue-1973.js

```diff
@@ -0,0 +1,96 @@
+// Copyright JS Foundation and other contributors, http://js.foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* issue #1973 */
+
+assert (777E7777777777 == Infinity)
+assert (-777E7777777777 == -Infinity)
+assert (777E-7777777777 == 0)
+assert (-777E-7777777777 == -0)
+
+assert (123E1000 == Infinity)
+assert (-123E1000 == -Infinity)
+assert (123E-1000 == 0)
+assert (-123E-1000 == -0)
+assert (123E00000000000 == 123)
+assert (123E000000000001 == 1230)
+
+assert (0.00000000001E301 == 1E+290)
+assert (0.0000000001E300 == 1E+290)
+assert (0.000000001E299 == 1E+290)
+
+assert (0.1E309 == 1E308)
+assert (0.1E310 == Infinity)
+assert (-0.1E309 == -1E308)
+assert (-0.1E310 == -Infinity)
+
+/* Testing numbers from 1e-324 to 1e-325 */
```

Comment:
all the for loops below should be removed. they do not test literal parsing at all. they test math implementation, but that should not happen here.